ui/src/main/resources/PhenoTips/VariantDisplay.xml

```diff
@@ -167,14 +167,15 @@
             this.optionsButton.update(new Element('a', {'class' : 'button', 'href' : '#'}).insert(this.Strings["hideOptionsButton"])).stopObserving('click', this.showOptions).observe('click', this.hideOptions.bind(this));
             
             var numberOfGenesInput = new Element('select', {'id' : 'variant-display-number-of-genes'});
-            for (var i = 1; i &lt;= 20; i++) {
+            var maxGenes = ("$xwiki.hasAccessLevel('edit')" == "true") ? 500 : 10;
```

Comment:
This is wrong for two reasons.

1. This checks edit access on the `PhenoTips.VariantDisplay` document, which will be false for everybody except admins.
2. The JS file is going to be cached, so the access level that happened to be available when first requesting the JS will be used until a cache refresh happens.